src/keyszer/config_api.py

```diff
@@ -150,33 +150,41 @@ class TypingTooLong(Exception):
 class UnicodeNumberToolarge(Exception):
     pass
 
+unicode_combo_list = []
+
 
 def to_US_keystrokes(s):
     """
-    Turn alphanumeric string (with spaces and some ASCII) up to length of 100 characters into keystroke commands
+    Turn alphanumeric string (with spaces and some ASCII) up to length 
+    of 100 characters into keystroke commands
 
     Warn: Almost certainly not going to work with non-US keymaps.
     """
-    if len(s) > 100:
-        raise TypingTooLong("`to_keystrokes` only supports strings of 100 characters or less")
-
-    combo_list = []
-    for c in s:
-        if ord(c) > 127:
-            combos = unicode_keystrokes(ord(c))
-            combo_list.extend(combos)
-        elif c.isupper():
-            combo_list.append(combo("Shift-" + c))
-        elif (str.isalnum(c)):
-            combo_list.append(Key[c.upper()])
-        elif c in ASCII_TO_KEY:
-            combo_list.append(ASCII_TO_KEY[c])
-        elif c in ASCII_WITH_SHIFT:
-            combo_list.append(ASCII_WITH_SHIFT[c])
-        else:
-            raise CharacterNotSupported(f"The character {c} is not supported by `to_keystrokes` yet.")
-
-    return combo_list
+    def _to_keystrokes(ctx):
+        if len(s) > 100:
+            raise TypingTooLong("`to_keystrokes` only supports strings of 100 characters or less")
+        combo_list = []
+        for c in s:
+            if ord(c) > 127:
+                unicode_keystrokes(ord(c))(ctx)    # extra "()" updates global variable
+                combo_list.extend(unicode_combo_list)
+            elif c.isupper():
+                if ctx.capslock_on: combo_list.append(combo(c))
```

Comment:
If caps lock is on, is this downcasing then?

Reply:
It's avoiding using a Shift keystroke with normal letters if CapLock is on, since normal letters will respond to the CapsLock state and be shifted to uppercase automatically.

Reply:
Elsewhere it's of course shifting lowercase normal letter characters, so that they actually come out lowercase when CapsLock is on.